server/tserver/src/main/java/org/apache/accumulo/tserver/logger/LogReader.java

```diff
@@ -123,12 +127,16 @@ public void execute(String[] args) throws Exception {
       Set<Integer> tabletIds = new HashSet<>();
 
       for (String file : opts.files) {
-
         Path path = new Path(file);
         LogFileKey key = new LogFileKey();
         LogFileValue value = new LogFileValue();
 
         if (fs.getFileStatus(path).isFile()) {
+          if (file.endsWith(".rf")) {
+            log.error(
+                "Can not read from a single rfile. Please pass in a directory for recovery logs.");
+            continue;
+          }
           // read log entries from a simple hdfs file
           try (final FSDataInputStream fsinput = fs.open(path);
               DataInputStream input = DfsLogger.getDecryptingStream(fsinput, siteConfig)) {
```

Comment:
Given this error message and new behavior, it's not clear what this condition (single file, but not ending in ".rf") is supposed to do. Is this for reading the older sequence files? A comment could go a long way here. The only comment it has is "read log entries from a simple hdfs file". Clearly, that can't be read from a simple RFile in HDFS, so that comment could be updated to make it clear what files it *can* read (i.e. what does "simple hdfs file" mean?).

Reply:
This area of the code is still used for wal-info to read in a single wal file (just tested to confirm). I will add additional comments because I agree, it is confusing.

Reply:
I moved the comment up and tried to be more explicit in my latest commit. Let me know if you think I should add more information to it.